forest/sycamore/survey_config.py

```diff
@@ -99,40 +146,84 @@ def gen_survey_schedule(config_path, time_start, time_end, beiwe_ids):
         time_end(str):
             The last date of the survey data
         beiwe_ids(list):
-            List of users in study for which we are generating a survey schedule
+            List of users in study for which we are generating a survey
+            schedule
+        all_interventions_dict(dict):
+            Dictionary containing a key for every beiwe id.
+            Each value in the dict is a dict, with a key for each intervention
+            and a timestamp for each intervention time
 
     Returns:
         times_sur(DataFrame):
-            DataFrame with a line for every survey deployed to every user in the study for the given time range
+            DataFrame with a line for every survey deployed to every user in
+            the study for the given time range
     """
     # List of surveys
     surveys = read_json(config_path)['surveys']
     # For each survey create a list of survey times
     times_sur = []
     for u_id in beiwe_ids:
         for i, s in enumerate(surveys):
+            s_times: list = []
             if s['timings']:
-                s_times = generate_survey_times(time_start, time_end, timings=s['timings'])
-                # Add in relative and absolute survey timings here
-                ###
-                tbl = pd.DataFrame(s_times, columns=['delivery_time'])
-                # Create the "next" time column too, which indicates the next time the survey will be deployed
-                tbl['next_delivery_time'] = tbl.delivery_time.shift(-1)
-                tbl['id'] = i
-                tbl['beiwe_id'] = u_id
-                # Get all question IDs for the survey
-                qs = [q['question_id'] for q in s['content'] if 'question_id' in q.keys()]
-                if len(qs) > 0:
-                    q_ids = pd.DataFrame({'question_id': qs})
-                    tbl = pd.merge(tbl, q_ids, how='cross')
-                times_sur.append(tbl)
-
-    times_sur = pd.concat(times_sur).reset_index(drop=True)
-    return times_sur
-
-
-def survey_submits(study_dir, config_path, time_start, time_end, beiwe_ids, agg, study_tz=None):
+                s_times = s_times + \
+                    generate_survey_times(
+                        time_start, time_end, timings=s['timings'],
```

Comment:
`generate_survey_times()` expects strings as the first two arguments, so need to either update the function signature or cast to string

Reply:
survey_submits was unnecessarily casting these as timestamps, so I deleted that and the conflict is removed.